### ISSUE TITLE
react-native-cli version check broken on Linux systems using GNU grep

### ISSUE THREAD

**Issue by progman32** [infinitered/ignite#259]
tl;dr need to pass `-P` to `grep`.

- [x] Issue does not already exist
- Identify exactly what aspect your issue is about
  - [ ] ignite base application
  - [x] ignite the cli
  - [ ] ignite generated code

On my Ubuntu system I run into this issue:
```
$ ignite new IgniteHelloWorld
🔥 Setting IgniteHelloWorld on FIRE! 🔥
<snip>
-----------------------------------------------
Igniting IgniteHelloWorld

𝗫 Must have at least version 1.x - 'npm install -g react-native-cli'
```

However, react-native-cli is for sure installed:
```
$ react-native -v
react-native-cli: 1.0.0
react-native: n/a - not inside a React Native project directory
```

The culprit is this check: https://github.com/infinitered/ignite/commit/1c47ccc91a7d7e7ebe2c0aa398d749b4cd0671e4#diff-7a6ed336074319f13867bb62262e9c3bR141

It's assuming grep will default to PCRE expressions, but GNU grep does not. Fix is to pass -P to grep. Demo:

```
$ react-native -v | grep 'react-native-cli: [1-9]\d*\.\d\.\d'
$ react-native -v | grep -P 'react-native-cli: [1-9]\d*\.\d\.\d'
react-native-cli: 1.0.0
```

- `node -v`: v4.4.7
- `npm -v`: 3.10.5
- `react-native -v` (cli): 1.0.0
- `react-native -v`: n/a
- `ignite --version`: 1.3.2

**Comment by progman32** [comment]
Hm, some searching suggests `-P` isn't supported on OSX. May be a better idea to just do the matching in JS instead of using grep.

**Comment by RootMale** [comment]
+1

**Comment by skellock** [comment]
Here's what I just did to dodge the strange version output of `react-native-cli`.  Perhaps we can plug this in if it checks out on ye olde linux.

https://github.com/infinitered/ignite/pull/264/files#diff-d8b6af7dfcff66d91d99d00b3a644ac3R96

**Comment by GantMan** [comment]
this is solved in 1.4.0
